### PR TITLE
[20722] Force unlimited ResourceLimits if lower or equal to zero

### LINKED ISSUE #4609: Allow inconsistent about DurabilityService QoS
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

When setting history_depth = 0 or max_samples = 0 in DurabilityService QoS, return bad parameter or inform inconsistent QoS.

### Current behavior

Despite inconsistent QoS settings, datareader, datawriter and topic are created normally.

### Steps to reproduce

It is inconsistent if `max_samples < (max_instances * max_samples_per_instance)` . But when setting max_instances=0 or history_depth=0, Despite inconsistent QoS settings, datareader, datawriter and topic are created normally.

Insert the following code in HelloWorldSubscriber.cpp in HelloWorldExample.

```cpp
// Creating DataReader QoS like this caused error.

    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
		rqos.durability_service().history_depth = 0;
		rqos.durability_service().max_samples = 0;
		
		DataWriterQos rqos = DATAWRITER_QOS_DEFAULT;
		rqos.durability_service().history_depth = 0;
		rqos.durability_service().max_samples = 0;
		
		TopicQoS tqos = TOPIC_QOS_DEFAULT;
		tqos.durability_service().history_depth = 0;
		tqos.durability_service().max_samples = 0;
```

### Fast DDS version/commit

`Fast-DDS v2.11.2`, but I think it's going to happen in the latest version, too

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

Default configuration, UDPv4 & SHM, Intra-process

### Additional context

This issue is similar with #4574

### XML configuration file

_No response_

### Relevant log output

_No response_

### Network traffic capture

_No response_

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR forces the ResourceLimitsQoS values (`max_samples`, `max_instances` and `max_samples_per_instance`) to be considered as unlimited if a value **greater than zero** is **NOT** provided.

@Mergifyio backport 2.13.x 2.10.x 2.6.x

Fixes #4609

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- [x] New feature has been documented/Current behavior is correctly described in the documentation. 
    Related documentation PR: eProsima/Fast-DDS-docs#738
- [x] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/mock/rtps/PublisherHistory/fastdds/publisher/DataWriterHistory.hpp [hunk 1]
```diff
@@ -79,12 +79,12 @@ class DataWriterHistory : public WriterHistory
         , topic_att_(topic_att)
         , unacknowledged_sample_removed_functor_(unack_sample_remove_functor)
     {
-        if (resource_limited_qos_.max_instances == 0)
```
Comment (EduPonz) [inline test/mock/rtps/PublisherHistory/fastdds/publisher/DataWriterHistory.hpp:82]: Same as in DataWriterHistory

File: test/mock/rtps/PublisherHistory/fastrtps/publisher/PublisherHistory.h [hunk 2]
```diff
@@ -74,12 +74,12 @@ class PublisherHistory : public WriterHistory
         , resource_limited_qos_(topic_att.resourceLimitsQos)
         , topic_att_(topic_att)
     {
-        if (resource_limited_qos_.max_instances == 0)
```
Comment (EduPonz) [inline test/mock/rtps/PublisherHistory/fastrtps/publisher/PublisherHistory.h:77]: Same as in PublisherHistory

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — CHANGES_REQUESTED** [review]
I think we need a test for creating the entities with negative values for resource limits

**Comment by EduPonz** [comment]
@Mergifyio backport 2.13.x 2.10.x 2.6.x

**Review by EduPonz — DISMISSED** [review]

**Comment by EduPonz** [comment]
@richiprosima please test this

**Review by EduPonz — APPROVED** [review]
